File: CWE-89.cs
using System;
using System.Data.SqlClient;

class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("Enter your username:");
        string username = Console.ReadLine();
        
        string connectionString = "Data Source=server;Initial Catalog=database;User ID=user;Password=password";
        string query = "SELECT * FROM Users WHERE Username = @username";

        using (SqlConnection connection = new SqlConnection(connectionString))
        {
            SqlCommand command = new SqlCommand(query, connection);
            command.Parameters.AddWithValue("@username", username);
            connection.Open();
            SqlDataReader reader = command.ExecuteReader();
            
            if (reader.HasRows)
            {
                Console.WriteLine("User found!");
            }
            else
            {
                Console.WriteLine("User not found.");
            }
            
            reader.Close();
        }
    }
}

File: CWE-89.py
import sqlite3

def unsafe_query(user_input):
    connection = sqlite3.connect('example.db')
    cursor = connection.cursor()
    
    # Уязвимый запрос
    query = f"SELECT * FROM users WHERE username = '{user_input}'"
    cursor.execute(query)
    
    results = cursor.fetchall()
    connection.close()
    
    return results

# Пример использования
user_input = "admin' --"
print(unsafe_query(user_input))
